src/core/internal/gc/impl/conservative/gc.d

```diff
@@ -213,10 +213,17 @@ class ConservativeGC : GC
         runLocked!(go, otherTime, numOthers)(gcx);
     }
 
+    debug (GC_RECURSIVE_LOCK) static bool lockedOnThisThread;
```

Comment:
Please add the new condition to the list of conditions at the top with a short description.

BTW: does this change really help? IIRC throwing involves capturing the stack which is saved to memory allocated from the GC, so a staticError is not really `@nogc`. Maybe this has changed in the mean time, though.

Reply:
Good idea, done.

Yep, it does seem to work! The tests I added make use of it to catch the `Error`.

Reply:
> IIRC throwing involves capturing the stack which is saved to memory allocated from the GC

Yeah, `OutofMemoryError` errors do not have a stack trace, which incidentally makes them difficult to debug. Room for improvement there...